src/target/riscv/riscv-013.c

```diff
@@ -1712,8 +1745,8 @@ static int examine(struct target *target)
 	}
 
 	riscv013_info_t *info = get_info(target);
-	/* TODO: This won't be true if there are multiple DMs. */
-	info->index = target->coreid;
+	/* Up to 2^20 harts can be connected to a single DM */
+	info->index = target->coreid & (RISCV_MAX_HARTS - 1);
```

Comment:
```suggestion
	info->index = target->coreid % RISCV_MAX_HARTS;
```

Seems sloppy to assume the constant is a power of 2, even though it is. The performance difference doesn't matter.

Reply:
Thanks